app_dart/lib/src/model/firestore/pr_test_runs.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:github/github.dart';
+import 'package:googleapis/firestore/v1.dart' as g;
+import 'package:meta/meta.dart';
+
+import 'base.dart';
+
+/// Records presubmit test execution.
+///
+/// For a given GitHub pull request, we store:
+/// ```
+/// /projects/flutter-dashboard/databases/cocoon/prTestRuns/
+///   document: <firebase unique id>
+///     pr:   <number>
+///     repo: <owner>/<name>
+///     sha:  <string>
+///     test: <string>
+///     runs: <array of PrTestRunResult objects>
+/// ```
+final class PrTestRuns extends AppDocument<PrTestRuns> {
+  /// Description of the document in Firestore.
+  static final metadata = AppDocumentMetadata<PrTestRuns>(
+    collectionId: 'prTestRuns',
+    fromDocument: PrTestRuns.fromDocument,
+  );
+
+  @override
+  AppDocumentMetadata<PrTestRuns> get runtimeMetadata => metadata;
+
+  /// Create [PrCheckRuns] from a Commit Document.
+  factory PrTestRuns.fromDocument(g.Document prCheckRunsDoc) {
+    return PrTestRuns._()
+      ..fields = prCheckRunsDoc.fields!
+      ..name = prCheckRunsDoc.name!;
+  }
+
+  // Indexes:
+  //
+  // - {pr, repo}  --> Used to jump from the Github UI, or provide a nice /flutter/flutter/1234 URL
+  // - {sha}       --> Used to lookup all of the tests for a given SHA
+  static const _fieldPrNumber = 'pr';
+  static const _fieldRepo = 'repo';
+  static const _fieldSha = 'sha';
+  static const _fieldTestName = 'test';
+  static const _fieldRuns = 'runs';
+
+  PrTestRuns._();
+
+  /// The pull-request number.
+  int get prNumber => int.parse(fields[_fieldPrNumber]!.integerValue!);
```

Comment:
non-blocking: 

frak; `app_dart/lib/src/model/common/firestore_extensions.dart` only has dart -> Value. We should totally extend values to do the int.parse() thing.